src/DcmTrajectoryGenerator.cpp

```diff
@@ -0,0 +1,652 @@
+  
+  // evaluate the new Single Support trajectory parameters
+  std::shared_ptr<GeneralSupportTrajectory> currentSingleSupport = nullptr;
+  currentSingleSupport = std::make_shared<SingleSupportTrajectory>(singleSupportStartTime, singleSupportEndTime,
+								   t0, singleSupportDuration, m_omega,
+								   zmp, nextDcmIos);
+  
+  // instantiate position and velocity boundary conditions vectors
+  iDynTree::Vector2 initPosition, initVelocity;
+  iDynTree::Vector2 endPosition, endVelocity;
+
+  // the begining time of the Double Support phase coinceds with the beginning of the next Single Support phase
+  double doubleSupportEndTime = std::get<0>(nextSubTrajectoryDomain);
+
+  // set the boundary conditions
+  if(!currentSingleSupport->getDcmPos(singleSupportEndTime, initPosition)){
+    std::cerr << "[DCM TRAJECTORY GENERATOR] Error when the position of the DCM in the next SS phase is evaluated." <<std::endl;
+    return false;
+  }
+  
+  if(!currentSingleSupport->getDcmVel(singleSupportEndTime, initVelocity)){
+    std::cerr << "[DCM TRAJECTORY GENERATOR] Error when the velocity of the DCM in the next SS phase is evaluated." <<std::endl;
+    return false;
+  }
+  
+  if(!nextSingleSupport->getDcmPos(doubleSupportEndTime, endPosition)){
+    std::cerr << "[DCM TRAJECTORY GENERATOR] Error when the position of the DCM in the previous SS phase is evaluated." <<std::endl;
+    return false;
+  }
+
+  if(!nextSingleSupport->getDcmVel(doubleSupportEndTime, endVelocity)){
+    std::cerr << "[DCM TRAJECTORY GENERATOR] Error when the velocity of the DCM in the previous SS phase is evaluated." <<std::endl;
+    return false;
+  }
+
+  // evaluate the new Double Support phase
+  double doubleSupportStartTime = singleSupportEndTime;
+  std::shared_ptr<GeneralSupportTrajectory> currentDoubleSupport = nullptr;
+
+  // check if pause features is active and if it is needed
+  if (m_pauseActive && (doubleSupportEndTime - doubleSupportStartTime > m_maxDoubleSupportDuration)){
+    // in this case the DS phase is splitted in three DS phase
+    // In the first one the DCM of the robot has to reach the center of the feet convex hull
+    // In the second one the robot has to stop (stance phase)
+    // In the third one the DCM has to reach the "endPosition" with the "endVelocity"
+
+    // evaluate the stance position constrain
+    iDynTree::Vector2 stancePosition, stanceVelocity;
+    stanceVelocity.zero();
+    iDynTree::toEigen(stancePosition) = (iDynTree::toEigen(zmp) + iDynTree::toEigen(nextSingleSupport_cast->getZmp())) / 2;
+
+    // evaluate the start and the end time of the Stance phase
+    double doubleSupportStanceStartTime = doubleSupportStartTime + m_nominalDoubleSupportDuration / 2;
+    double doubleSupportStanceEndTime = doubleSupportEndTime - m_nominalDoubleSupportDuration / 2;
+
+    if(fabs(doubleSupportStanceStartTime - doubleSupportStanceEndTime) < MIN_DURATION_DOUBLE_SUPPORT_STANCE_PHASE){
```

Comment:
This line is not very clear to me. First I don't see the necessity of ``fabs``, since ``m_nominalDoubleSupportDuration`` should be lower than ``m_maxDoubleSupportDuration``. This should be checked when receiving those inputs. In addition ``m_nominalDoubleSupportDuration`` should be greater than ``MIN_DURATION_DOUBLE_SUPPORT_STANCE_PHASE`` and this should be checked again when those values are obtained.